src/truecallerpy/secrets_provider.py

```diff
@@ -0,0 +1,78 @@
+import json
+import git
+import os
+
+from google.cloud.secretmanager import (
+    AccessSecretVersionResponse,
+    SecretManagerServiceClient,
+)
+
+SECRETS_VERSION = 7
+GLOBAL_SECRET_NAME = "env_file"
+GOOGLE_PROJECT_NAME = "datalane"
+
+
+class NoSecretFoundError(Exception):
+    """Throws when no secret is present in SecretManager"""
+
+
+def get_datalane_sa_key_file_path() -> str:
+    """Get Datalane Service account secret location"""
+    # Get the current directory
+    current_dir = os.getcwd()
+
+    # Find the root of the Git repository - this will work for truecallerpy since it'll look for the datalane repo's root location
```

Comment:
Have you had a chance to test this?

I.e when we are running from our main repo and then have truecallerpy as a package, does this successfully find the datalane root?

My immediate thought is, does it find it's own root (the root of truecallerpy) or the datalane root

Reply:
I've tested this locally by re-creating a new poetry.lock inside the datalane repo to point to this branch and it successfully made the true callers calls